state_transfer/transport/transport.go

```diff
@@ -28,6 +28,18 @@ type Transport interface {
 	Direct() bool
 	CreateServer(client.Client, endpoint.Endpoint) error
 	CreateClient(client.Client, endpoint.Endpoint) error
+	ProxyOptions() *ProxyOptions
+	SetProxyOptions(*ProxyOptions)
```

Comment:
We have removed the Setxxxx options from this interface, I wonder if we can consider removing this and just adding it as a constructor? 

Should we consider a larger "Options" struct that only contains ProxyOptions right now so that we can add in the future options without having to create more methods?

Reply:
Yep. I don't believe we need any of the setters. I'll get rid of them. Remember now we were dumping all of them.

@pranavgaikwad I can turn ProxyOptions into a generic Options and add the ca verify stuff there too, perhaps. WDYT? Are those settings relevant outside of proxy support? I seem to think they are, but I'm guessing based off past conversations.

Reply:
@jmontleon those settings are not relevant outside of proxy. And yeah, I agree to Shawn's point that it makes sense to abstract them as options instead because not all transports may have the same settings as Stunnel. Apart from that, the settings themselves are correct and should work.